actionpack/test/controller/render_test.rb

```diff
@@ -323,11 +323,12 @@ def setup
   end
 
   def test_dynamic_render_with_file
-    # This is extremely bad, but should be possible to do.
     assert File.exist?(File.expand_path("../../test/abstract_unit.rb", __dir__))
-    response = assert_deprecated { get :dynamic_render_with_file, params: { id: '../\\../test/abstract_unit.rb' } }
-    assert_equal File.read(File.expand_path("../../test/abstract_unit.rb", __dir__)),
-      response.body
+    assert_deprecated do
+      assert_raises ActionView::MissingTemplate do
+        get :dynamic_render_with_file, params: { id: '../\\../test/abstract_unit.rb' }
+      end
+    end
```

Comment:
This test best illustrates the change (though it confused me for a bit). It first asserts that `../../test/abstract_unit.rb` exists relative to **this file's_ directory**, `__dir__`. ie. ` actionpack/test/controller/../../test/abstract_unit.rb`.

When rendering, however, it used to succeed by finding `../../test/abstract_unit.rb` relative to the view_path. ie. `actionpack/test/fixtures/../../test/abstract_unit.rb`.

I don't think anyone (other than this test) does this, and don't think it should be possible (though it isn't an immediate security issue, we already expect `render file:` to be allowed to render anything on the FS).